File: frontend/src/lib/platforms.ts
import type { PlatformMeta } from "../types";

export const fallbackPlatforms: PlatformMeta[] = [
  { id: "xhs", name_cn: "小红书", name_en: "XiaoHongShu", enabled: true, status: "enabled", accent_color: "#ff2442", icon: "xhs" },
  { id: "douyin", name_cn: "抖音", name_en: "Douyin", enabled: false, status: "coming_soon", accent_color: "#111111", icon: "douyin" },
  { id: "kuaishou", name_cn: "快手", name_en: "Kuaishou", enabled: false, status: "coming_soon", accent_color: "#ff7a00", icon: "kuaishou" },
  { id: "weibo", name_cn: "微博", name_en: "Weibo", enabled: false, status: "coming_soon", accent_color: "#e6162d", icon: "weibo" },
  { id: "xianyu", name_cn: "闲鱼", name_en: "Xianyu", enabled: false, status: "coming_soon", accent_color: "#ffe100", icon: "xianyu" },
  { id: "taobao", name_cn: "淘宝", name_en: "Taobao", enabled: false, status: "coming_soon", accent_color: "#ff5000", icon: "taobao" }
];

export function getPlatform(id: string | undefined) {
  return fallbackPlatforms.find((platform) => platform.id === id) ?? fallbackPlatforms[0];
}

File: backend/app/core/platforms.py
from __future__ import annotations

from dataclasses import asdict, dataclass
from enum import Enum
from typing import List


class PlatformId(str, Enum):
    XHS = "xhs"
    DOUYIN = "douyin"
    KUAISHOU = "kuaishou"
    WEIBO = "weibo"
    XIANYU = "xianyu"
    TAOBAO = "taobao"


@dataclass(frozen=True)
class PlatformMeta:
    id: PlatformId
    name_cn: str
    name_en: str
    enabled: bool
    status: str
    accent_color: str
    icon: str

    def to_dict(self) -> dict:
        payload = asdict(self)
        payload["id"] = self.id.value
        return payload


_PLATFORMS: List[PlatformMeta] = [
    PlatformMeta(PlatformId.XHS, "小红书", "XiaoHongShu", True, "enabled", "#ff2442", "xhs"),
    PlatformMeta(PlatformId.DOUYIN, "抖音", "Douyin", False, "coming_soon", "#111111", "douyin"),
    PlatformMeta(PlatformId.KUAISHOU, "快手", "Kuaishou", False, "coming_soon", "#ff7a00", "kuaishou"),
    PlatformMeta(PlatformId.WEIBO, "微博", "Weibo", False, "coming_soon", "#e6162d", "weibo"),
    PlatformMeta(PlatformId.XIANYU, "闲鱼", "Xianyu", False, "coming_soon", "#ffe100", "xianyu"),
    PlatformMeta(PlatformId.TAOBAO, "淘宝", "Taobao", False, "coming_soon", "#ff5000", "taobao"),
]


def get_platforms() -> List[PlatformMeta]:
    return list(_PLATFORMS)


def get_platform(platform_id: PlatformId) -> PlatformMeta:
    for platform in _PLATFORMS:
        if platform.id == platform_id:
            return platform
    raise KeyError(platform_id)
